xrviz/dashboard.py

```diff
@@ -0,0 +1,93 @@
+
+
+class Dashboard(SigSlot):
+    """
+    This section provides access to the complete generated dashboard,
+    consisting of all subsections.
+
+    Parameters
+    ----------
+    data: `xarray` instance: `DataSet` or `DataArray`
+           datset is used to initialize.
+
+    Attributes
+    ----------
+    panel: Displays the generated dashboard.
+    control: Provides access to the generated control panel.
+    index_selectors: Provides access to generated index_selectors.
+    plot: Plot button, upon click generates graph according to
+          kwargs selected in other sub-sections.
+    output: Provides access to generated graph.
+    """
+
+    def __init__(self, data):
+        super().__init__()
+        self.data = data
+        self.control = Control(self.data)
+        self.plot = pn.widgets.Button(name='Plot', width=200)
+        self.index_selectors = self.control.fields.index_selectors
+        self.output = pn.Row(pn.Spacer(name='Graph'))
+
+        self._register(self.plot, 'plot_clicked', 'clicks')
+        self.connect('plot_clicked', self.create_plot)
+        self.control.displayer.connect('variable_selected', self.check_is_plottable)
+
+        self.panel = pn.Column(self.control.panel,
+                               self.plot,
+                               self.output
+                               )
+        if isinstance(self.data, xr.Dataset):
+            self.plot.disabled = True
+        else:
+            if self.data.name in self.data.coords or len(self.data.dims) <= 1:
+                self.plot.disabled = True
+
+    def create_plot(self, *args):
+        kwargs = self.control.kwargs
+        var = kwargs['']
+        graph_opts = {}
+        graph_opts['x'] = kwargs['x']
+        graph_opts['y'] = kwargs['y']
+        graph_opts['rasterize'] = True
+        graph_opts['width'] = 600
+        graph_opts['height'] = 400
+        graph_opts['crs'] = ccrs.PlateCarree()
+
+        if isinstance(self.data, xr.Dataset):
+            graph_opts['title'] = var
+            self.graph = pn.Row(self.data[var].hvplot.quadmesh(**graph_opts))
+        else:
+            graph_opts['title'] = self.data.name
+            self.graph = pn.Row(self.data.hvplot.quadmesh(**graph_opts))
+
+        self.output[0] = self.graph[0][0]
+        self.fill_index_selectors()
+
+    def fill_index_selectors(self):
+        """
+        To convert the auto-generated slider, into Select and Player
+        widget.
+        """
+        self.index_selectors.clear()
+        try:
+            for widget in self.graph[0][1]:
+                selector = convert_widget(widget, pn.widgets.Select())
+                player = convert_widget(selector, pn.widgets.DiscretePlayer())
+                combined = pn.Column(selector, player)
+                self.index_selectors.append(combined)
+        except IndexError as e:
+            pass
+
+    def check_is_plottable(self, var):
+        var = var[0]
+        self.plot.disabled = False
```

Comment:
There should be no need to set this if we are about to set it to True in a couple of lines. This feels like an `else`.

Reply:
Without setting this Plot button will remain disabled forever. Each time a variable is selected in case of Dataset, the button is enabled. It is only disabled if variable does not satisfies the conditions.